File: android/app/src/main/java/com/smartexpense/android/domain/usecase/notification/MarkAllNotificationsReadUseCase.java
package com.smartexpense.android.domain.usecase.notification;

import com.smartexpense.android.domain.repository.NotificationRepository;
import com.smartexpense.android.domain.repository.ResultCallback;

public class MarkAllNotificationsReadUseCase {
    private final NotificationRepository repository;

    public MarkAllNotificationsReadUseCase(NotificationRepository repository) {
        this.repository = repository;
    }

    public void execute(ResultCallback<Void> callback) {
        repository.markAllAsRead(callback);
    }
}

File: android/app/src/main/kotlin/com/smartexpense/android/domain/usecase/notification/MarkAllNotificationsReadUseCase.kt
package com.smartexpense.android.domain.usecase.notification

import com.smartexpense.android.domain.repository.NotificationRepository

class MarkAllNotificationsReadUseCase(private val repository: NotificationRepository) {
    suspend fun execute(): Result<Unit> {
        return repository.markAllAsRead()
    }
}
